src/Uno.UI/UI/Xaml/Input/FocusManager.wasm.cs

```diff
@@ -40,9 +42,28 @@ internal static void ProcessElementFocused(UIElement element)
 				_log.Value.LogDebug($"{nameof(ProcessElementFocused)}() focusedElement={GetFocusedElement()}, element={element}, searching for focusable parent control");
 			}
 
-			// Try to find the first focusable parent and set it as focused, otherwise just keep it for reference (GetFocusedElement())
-			var ownerControl = element.GetParents().OfType<Control>().Where(control => control.IsFocusable).FirstOrDefault();
-			ProcessControlFocused(ownerControl);
+			foreach (var parent in element.GetParents())
+			{
+				_log.Value.LogDebug($"Checking {parent}. TextBlock.IsFocusable: {(parent as TextBlock)?.IsFocusable == true}");
```

Comment:
This looks like it's going to be executed every time

Reply:
Ah, forgot this line there...